web3/eth.py

```diff
@@ -104,7 +104,23 @@
 )
 
 
-class Eth(Module):
+class BaseEth(Module):
+    _gas_price: Method[Callable[[], Wei]] = Method(
+        RPC.eth_gasPrice,
+        mungers=None,
+    )
+
+
+class AsyncEth(BaseEth):
+    is_async = True
+
+    @property
+    async def gas_price(self) -> Wei:
+        # types ignored b/c mypy conflict with BlockingEth properties
```

Comment:
Yeah, this is unfortunate and would be nice to resolve... but I'm not sure what to suggest.